### PR TITLE
Various config refactors and changes to be cleaner and easier to unde…

### PR DESCRIPTION
**Pull request by buffet-time**
## What
To bring the repo to what was discussed with config files being preferred to be Json > Ts > Js.

Refactored config files to be in this style, like Eslint from js to json jest from js to ts, and leaving tailwind as js due to lack of TS support. 

Also updated eslint config a bit to remove some redundancy and to ensure proper rule coverage to catch things like strict null usage.

### CODE REVIEW

File: nuxt.config.ts [hunk 1]
```diff
@@ -47,6 +47,4 @@ const config: NuxtConfig = {
       lang: 'en',
     },
   },
-};
-
-export default config;
+} as NuxtConfig;
```
Comment (legrandgrand) [inline nuxt.config.ts:50]: Why do you have to use as here? Would be better to keep the previous version

Comment (buffet-time) [reply nuxt.config.ts:50]: I did that so we're not creating a variable which isn't needed. So if we just default export the object and slap as NuxtConfig it still gets correct typing for linting and autocomplete

File: tailwind.config.js [hunk 2]
```diff
@@ -1,5 +1,4 @@
-const { screens } = require('./breakpoints.js');
-
+/* eslint-disable vue/sort-keys */
```
Comment (zysim) [inline tailwind.config.js:1]: Why did we remove `breakpoints.js` here?

Comment (buffet-time) [reply tailwind.config.js:1]: I didn't see it being used anywhere else, and logically if someone needs these values they can be accessed from that config file if needed.

**Review by erunks — COMMENTED** [review]
Code looks good, just need to push the updated snapshots to get CI passing

**Comment by zysim** [comment]
Shit's cash now :sunglasses:

**Review by legrandgrand — CHANGES_REQUESTED** [review]
can you please remove the as any/ as [type] in the code? this is bad typescript practices as we force a type onto a variable instead of checking that the variable is a certain type

File: jest.setup.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  }
+
+  // remove .vue from the last element
+  componentRoutes[componentRoutes.length - 1] = routeValue.replace('.vue', '');
+  componentRoutes[0] = upperFirst(componentRoutes[0]);
+
+  return componentRoutes;
+}
+
+global.Vue = Vue as any;
```
[the remark this answers is not in the document]
Comment (buffet-time) [reply jest.setup.ts:49]: The types VyeConstructor and Vue don't perfectly overlap. It seems like for our current purposes this is fine so casting to Any is fine

[the remark this answers is not in the document]
Comment (buffet-time) [reply jest.setup.ts:49]: And regarding the as NuxtConfig thats actually untrue. If you untype and dont do any as linters won't show you anything but any types. Do as Nuxt and it's typed properly as NuxtConfig.

**Review by legrandgrand — APPROVED** [review]

File: jest.setup.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (legrandgrand) [reply jest.setup.ts:49]: On that last point I am not convinced in general but I will trust you for that one :)

**Review by erunks — APPROVED** [review]

**Pull request merged**